PROVESFlightControllerReference/Components/StackMonitor/StackMonitor.cpp

```
@@ -0,0 +1,79 @@
+
+//! k_thread_foreach callback: append this thread's stack usage to the
+//! ThreadStackSampleSet pointed to by userData. Zephyr runs this callback
+//! with the thread-monitor spinlock held and IRQs locked, so it must not
+//! allocate or block: it only reads the stack high-water mark, does a
+//! bounded string copy of the name, and stores integers into a fixed slot
+//! (ThreadStackSampleSet::add). Keeps all Zephyr calls out of
+//! StackMonitorCore, which stays pure.
+void appendThreadSample(const struct k_thread* thread, void* userData) {
+    auto* samples = static_cast<ThreadStackSampleSet*>(userData);
+
+    std::size_t unusedBytes = 0;
```

Clang (14.0.6)

[warning] variable 'unusedBytes' is not initialized

(cppcoreguidelines-init-variables)